src/utils/definition.js

```diff
@@ -30,4 +31,32 @@ export default class Definition {
     const effective = Math.ceil((get(definition, 'described.score', 0) + get(definition, 'licensed.score', 0)) / 2)
     return { tool, effective }
   }
+
+  /**
+   * Revert a list of definitions or a specific one, removing all the changes or only specific values
+   * @param  {[]} components list of definitions
+   * @param  {{}} definition specific definition, if null the function will check all the definitions
+   * @param  {string} value string that identifies the specific value to revert, if null all the changes will be removed
+   */
+  static revert(components, definition, value) {
+    if (!components) return
+    return components.map(component => {
+      const withoutChanges = this.revertChanges(component, value)
+      if (!definition) return withoutChanges
+      else if (isEqual(EntitySpec.fromCoordinates(definition), EntitySpec.fromCoordinates(component)))
+        return withoutChanges
+      else return component
+    })
+  }
+
+  static revertChanges(component, value) {
+    if (value) {
+      const { [value]: omit, ...withoutChanges } = component.changes
+      component.changes = withoutChanges
+      return component
+    } else {
+      const { changes, ...withoutChanges } = component
```

Comment:
The goal here is actually to preserve the initial object, instead of deleting property from it.
Given that, i found the spread operator + destructuring a much more effective solution.

I made some tests before posting an aswer.
Actually, this implementation of spread operator + destructuring, doesn't mutate the original object.
`const { changes, ...withoutChanges } = component` creates a new object `withoutChanges` which contains everything from `component` except the property `changes`.

I've put 2 console.log in the code as proof:
<img width="446" alt="schermata 2018-09-26 alle 11 24 58" src="https://user-images.githubusercontent.com/7654979/46070766-dfd33100-c17e-11e8-8988-5a64540b47fa.png">
and this is the result:
<img width="1424" alt="schermata 2018-09-26 alle 11 24 47" src="https://user-images.githubusercontent.com/7654979/46070758-d9dd5000-c17e-11e8-960a-bec0aadfbfdf.png">

As you can see, the component object has not been mutated.

Since the purpose is to not change the values in the components object, but instead return a new object and then using it to update the redux store, this would not affect any other use of the components object.

Makes sense?